src/lib/import_export/binary/binary_writer.hpp

```diff
@@ -223,6 +224,38 @@ class BinaryWriter {
   template <typename T>
   static void _write_segment(const LZ4Segment<T>& lz4_segment, bool column_is_nullable, std::ofstream& ofstream);
 
+  /**
+   * FSSTSegments are dumped with the following layout:
+   *
+   * Description                 | Type                                | Size in bytes
+   * --------------------------------------------------------------------------------------------------------
+   * Encoding Type               | EncodingType                        | 1
+   * Attribute vector compr. ID. | CompressedVectorTypeID              | 1
+   * Compressed values size      | uint32_t                            | 4
+   * Compressed values           | vector<uint32_t>                    | Compressed values size * 1
+   * Vector compress. bit width  | uint8_t                             | 1
+   * Offset values               | uint8_t                             | (Rows + 1) * (vector compr. bit width) / 8
+   *                                                                     rounded up to next multiple of word (8 byte)
+   * Offset values               | uint(8|16|32)_t                     | (Rows + 1) * width of offset vector
+   * Reference offsets size      | uint32_t                            | 4
+   * Reference offsets           | vector<uint64_t>                    | 8 * 8
+   * NULL values size            | uint32_t                            | 4
+   * NULL values¹                | vector<bool> (BoolAsByteType)       | size * 1
+   * # elements in a bucket      | uint64_t                            | 8
```

Comment:
Not that the bits matter, but do we really need a uint64_t here?

Reply:
Not really. We've already changed it to `uint32_t`.